test/test_arhp_with_non_legacy_connection_handling.rb

```diff
@@ -56,7 +56,7 @@ def test_correctly_writes_to_sharded_databases_and_only_switches_dbs_when_necess
       # Assert that we switched, and only switched, in the order we expected.
       # If this assertion starts to fail, Rails is likely calling `#connection`
       # somewhere new, and we should investigate
-      db_switches = new_logger.string.scan(/select_db (\w+)/).flatten
+      db_switches = new_logger.string.scan(/[\bselect\b|\bchange\b]_db (\w+)/).flatten
```

Comment:
I believe `[]` is for characters ranges, while here we want to search for whole words. This might work better: (the `?:` avoid making a capture from the `()` group)
```suggestion
      db_switches = new_logger.string.scan(/(?:select_db|change_db) (\w+)/).flatten
```